docs/API/V1/server_config.md

```diff
@@ -0,0 +1,253 @@
+[metadata](../metadata.md) key `server.deletion` is calculated based on this
+value and user preferences (which may specify a shorter lifetime, but not a
+longer lifetime). When a dataset has remained on the server past the
+`server.deletion` date, it may be removed automatically by the server to
+conserve space.
+
+The number of days is specified as an string representing an integer, optionally
+followed by a space and `day` or `days`. For example, "4" or "4 days" or "4 day"
+are equivalent.
+
+```
+{
+    "dataset-lifetime": "4"
+}
+```
+
+### server-banner
+
+This server configuration setting allows a server administrator to set an
+informational message that can be retrieved and displayed by any client, for
+example as a banner on a UI. The value s a JSON object, containing at least
+a `message` field.
+
+Any additional JSON data may be provided, and will be returned uninterpreted
+when a client requests it with `GET /api/v1/server/configuration/server-banner`.
+
+For example,
+
+```
+{
+    "server-banner": {
+        "message": "The server will be down for 2 hours on Monday, July 41st",
+        "contact": {
+            "email": "admin@pbench.example.com",
+            "phone": "(555) 555-5555",
+            "hours": ["8:00 EST", "17:00 EST"]
+        },
+        "statistics": {
+            "datasets": 50000,
+            "hours-up": 2.3,
+            "users": 26.4
+        }
+    }
+}
+```
+
+### server-state
+
+This server configuration setting allows a server administrator to control
+the operating state of the server remotely. As for
+[`server-banner`](#server-banner), the value is a JSON object, and any JSON
+fields passed in to the server will be returned to a client, however the
+following fields have special meaning:
+
+**`status`** \
+The operating status of the server.
+
+* `enabled`: Normal operation.
+* `disabled`: Most server API endpoints will fail with the **503** (service
+unavailable) HTTP status. However a few endpoints are always allowed:
+  * [login](./login.md) because only an authenticated user with `ADMIN` role
+    can modify server configuration settings;
+  * [logout](./logout.md) for consistency;
+  * [endpoints](./endpoints.md) for server configuration information;
+  * [server_configuration](./server_config.md) to allow re-enabling the server
+    or modifying the banner.
+* `readonly`: The server will respond to `GET` requests for information, but will return **503** (service unavailable) for any attempt to modify server state. (With the same exceptions as listed above for `disabled`.)
+
+**`message`** \
+A message to explain downtime. This is required when the `status` is `disabled`
+or `readonly` and optional otherwise. Note that a `message` when the `status`
+is `enabled` won't normally be seen, unless a client asks for the `server-state`
+configuration setting. The `server-banner` configuration setting is generally
+more appropriate under normal operating conditions.
```

Comment:
I don't understand what "seen" means in this context.  The user isn't going to "see" anything except when the client fetches it from the server and displays it, and that includes the banner as well as this message.

The value associated with the `message` key is part of the JSON object set/retrieved for the `server-state` configuration key.  The `message` (if set) will (I think) always be available to the client in the same contexts where the `status` value is available...so, yeah, if the client doesn't request the `status` it's not going to get the `message`, and (unless we're supporting "dot-paths" here...are we?) the client isn't going to fetch the `status` without also receiving the `message` if there is one.  (Right?)

My instinct is that the banner is mostly for advertising the identity of the server (and perhaps for welcome messages and stuff like contact information), which doesn't change with operating state, while the message here is for temporary, status-related messages, which might include things like warnings of pending maintenance or service restoration announcements, which aren't really the fixed-sorts of things which would appear in the banner.

Reply:
Ah...re "seen", you're referring to the `503` response for `disabled` and `readonly`.  Maybe that should be the way this section is structured:

- In the event of a `503` response, the JSON object associated with `server-state` will be returned, which will allow the client to report the `message` to the user.
- Thus, the message associated with a `status` of `enabled` will only be seen by a user if the client explicitly fetches the `server-state` and displays it.
- And, therefore, while messages can be associated with a `status` of `enabled`, such messages are more likely to be seen by users if they are included in the banner `message`.